src/mlpack/methods/ann/layer/mean_pooling.hpp

```diff
@@ -14,23 +15,35 @@
 #define MLPACK_METHODS_ANN_LAYER_MEAN_POOLING_HPP
 
 #include <mlpack/prereqs.hpp>
+#include "layer.hpp"
 
 namespace mlpack {
 namespace ann /** Artificial Neural Network. */ {
 
+class MeanPoolingRule
+{
+ public:
+  /**
+   * Return the maximum value within the receptive block.
+   *
+   * @param input Input used to perform the pooling operation.  Could be an
+   *     Armadillo subview.
+   */
+  template<typename MatType>
+  typename MatType::elem_type Pooling(const MatType& input)
```

Comment:
You could mark this `static`, then you don't need to hold a `MeanPoolingRule` internally.  Honestly you could also just bring the `Pooling()` operation into the `MeanPoolingType` class; I don't see any benefit to this class.

Reply:
Done.